src/esssans/normalization.py

```diff
@@ -152,17 +161,51 @@ def iofq_denominator(
     # solid_angle (pixel-dependent) and monitors (wavelength-dependent) will fail.
     # The direct beam may also be pixel-dependent. In this case we need to drop
     # variances of the other term already before multiplying by solid_angle.
-    broadcast = {
-        UncertaintyBroadcastMode.drop: drop_variances_if_broadcast,
-        UncertaintyBroadcastMode.upper_bound: broadcast_with_upper_bound_variances,
-        UncertaintyBroadcastMode.fail: lambda x, sizes: x,
-    }[uncertainties]
     if direct_beam is not None:
+        broadcast = _broadcasters[uncertainties]
         denominator = direct_beam * broadcast(denominator, sizes=direct_beam.sizes)
-    denominator = solid_angle * broadcast(denominator, sizes=solid_angle.sizes)
     # Convert wavelength coordinate to midpoints for future histogramming
     # if wavelength_to_midpoints:
     denominator.coords['wavelength'] = sc.midpoints(denominator.coords['wavelength'])
+    return NormWavelengthTerm[RunType](denominator)
+
+
+def iofq_denominator(
+    wavelength_term: NormWavelengthTerm[RunType],
+    solid_angle: SolidAngle[RunType],
+    uncertainties: UncertaintyBroadcastMode,
+) -> Clean[RunType, Denominator]:
+    """
+    Compute the denominator term for the I(Q) normalization.
+
+    This is basically:
+    ``solid_angle * direct_beam * data_transmission_monitor * direct_incident_monitor / direct_transmission_monitor``
+    The `wavelength_term` included all but the `solid_angle` and is computed by
+    :py:func:`iofq_norm_wavelength_term`.
+
+    Because the multiplication between the wavelength dependent terms (monitor counts)
+    and the pixel dependent term (solid angle) consists of a broadcast operation which
+    would introduce correlations, we strip the data of variances.
```

Comment:
> we strip the data of variances.

I guess this is not always true, right? Since there is a mode for computing an upper bound?

Reply:
Yes, I updated it. The reference is not in the docstring of `UncertaintyBroadcastMode`, so I removed it here.